addons/stock/static/src/js/lazy_column_list.js

```diff
@@ -0,0 +1,187 @@
+     * (`_searchReadUngroupedList`), and let the controller manage it.
+     *
+     * @param {Object} element an element from the localData
+     * @param {Object} [options]
+     * @param {Object} [options.viewType] current viewType. If not set, we will
+     *   assume main viewType from the record
+     * @returns {string[]} the list of field names
+     */
+    _getFieldNames: function (element, options) {
+        const res = this._super.apply(this, arguments);
+        const fieldsInfo = element.fieldsInfo;
+        const viewType = (options && options.viewType) || element.viewType;
+        const lazyColumns = _getLazyColumns(fieldsInfo[viewType]);
+        return res.filter((column) => !lazyColumns.includes(column));
+    },
+
+    _makeDataPoint: function () {
+        // Should put the lazy column to work with decoration
+        const dataPoint = this._super.apply(this, arguments);
+        if (dataPoint.type === "record" && dataPoint.fieldsInfo) {
+            const fieldsInfo = dataPoint.fieldsInfo[dataPoint.viewType];
+            const lazyColumn = _getLazyColumns(fieldsInfo);
+            _bindLazyColumnToState(dataPoint, lazyColumn);
+        }
+        return dataPoint;
+    },
+});
+
+const LazyColumnListRenderer = ListRenderer.extend({
+    init: function () {
+        this.currentNbLazyRpc = 0;
+        return this._super.apply(this, arguments);
+    },
+
+    _lazyLoad: async function (state, lazyColumns) {
+        const res = await this._rpc(
+            {
+                model: state.model,
+                method: "read",
+                args: [state.res_ids, lazyColumns],
+            },
+            { shadow: true }
+        );
+        const resById = {};
+        for (const element of res.values()) {
+            resById[element.id] = {};
+            for (const lazyCol of lazyColumns) {
+                resById[element.id][lazyCol] = element[lazyCol];
+            }
+        }
+        _bindLazyColumnToState(this.state, lazyColumns, resById);
+        this.currentNbLazyRpc -= 1;
+        // Render only thr
+        for (const lazyCol of lazyColumns) {
+            const node = this.columns.find((colAttribute) => colAttribute.attrs.name === lazyCol);
+            const indexNode = this.columns.findIndex(
+                (colAttribute) => colAttribute.attrs.name === lazyCol
+            );
+            if (!node) {
+                continue;
+            }
+            const $columnHeader = this.$el.find(`th[data-name='${lazyCol}']`);
+            $columnHeader.replaceWith(this._renderHeaderCell(node));
+            const $cellsLazy = this.$el.find(`.o_data_cell[name='${lazyCol}']`);
+            for (let i = 0; i <= $cellsLazy.length; i++) {
+                let $cell = $($cellsLazy[i]);
+                let idRecord = $cell.parent() && $cell.parent().data("id");
+                let record = _findRecordsInState(this.state).find((rec) => rec.id === idRecord);
```

Comment:
As `$.parent()`  returns a jQuery object (which has a truly value) even if there is no parent, it's not necessary to check if `parent` returns something
```suggestion
                const $cell = $($cellsLazy[i]);
                const idRecord = $cell.parent().data("id");
                const record = _findRecordsInState(this.state).find((rec) => rec.id === idRecord);
```